py/htm/examples/encoders/simhash_document_encoder.py

```diff
@@ -0,0 +1,153 @@
+# ------------------------------------------------------------------------------
+# HTM Community Edition of NuPIC
+# Copyright (C) 2019, David McDougall
+#               2019, Brev Patterson, Lux Rota LLC, https://luxrota.com
+#
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the GNU Affero Public License version 3 as published by
+# the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero Public License for
+# more details.
+#
+# You should have received a copy of the GNU Affero Public License along with
+# this program.  If not, see http://www.gnu.org/licenses.
+# ------------------------------------------------------------------------------
+
+import htm.bindings.encoders
+SimHashDocumentEncoder = htm.bindings.encoders.SimHashDocumentEncoder
+SimHashDocumentEncoderParameters = htm.bindings.encoders.SimHashDocumentEncoderParameters
+
+
+if __name__ == '__main__':
+  """
+  Simple program to examine the SimHashDocumentEncoder.
+
+  For help using this program run:
+  $ python -m htm.examples.encoders.simhash_document_encoder --help
+  """
+  import argparse
+  import numpy
+  import random
+  import sys
+  import textwrap
+
+  from htm.bindings.sdr import SDR, Metrics
+
+  # Gather input from the user.
+  parser = argparse.ArgumentParser(
+      formatter_class=argparse.RawDescriptionHelpFormatter,
+      description="Simple program to examine the SimHashDocumentEncoder.\n\n" +
+      textwrap.dedent(SimHashDocumentEncoder.__doc__ + "\n\n" +
+                      SimHashDocumentEncoderParameters.__doc__))
+  parser.add_argument('--activeBits', type=int, default=0,
+                      help=SimHashDocumentEncoderParameters.activeBits.__doc__)
```

Comment:
The split py encoders/examples might move little too much. It is OK for me, I'll ask @ctrl-z-9000-times if he minds the arg parsing is moved from the RDSE encoder?

Reply:
That's fine, as long as you tested that the example still works as expected.

Reply:
Yup, working great still.

The ScalarEncoder already had all of that code in examples/.  RDSE had it in encoders/.  I had followed the RDSE, which @breznak asked to be moved into examples/. I didn't want to leave RDSE in different state than Scalar/SimHash, just wanted them to match.